pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java

```diff
@@ -323,6 +327,39 @@ public void testSendTimeout(int batchMessageDelayMs) throws Exception {
         assertEquals(cStat.getTotalMsgsReceived(), cStat.getTotalAcksSent());
         log.info("-- Exiting {} test --", methodName);
     }
+    
+    @Test
+    public void testBatchMessagesRateOut() throws PulsarClientException, InterruptedException, PulsarAdminException {
+        log.info("-- Starting {} test --", methodName);
+        String topicName = "persistent://my-property/cluster/my-ns/testBatchMessagesRateOut";
+        double produceRate = 10;
+        ConsumerConfiguration consumerConf = new ConsumerConfiguration();
+        consumerConf.setSubscriptionType(SubscriptionType.Exclusive);
+        Consumer consumer = pulsarClient.subscribe(topicName, "my-subscriber-name", consumerConf);
+        ProducerConfiguration producerConf = new ProducerConfiguration();
+        producerConf.setBatchingMaxMessages(5);
+        producerConf.setBatchingEnabled(true);
+        producerConf.setBatchingMaxPublishDelay(2, TimeUnit.SECONDS);
+
+        Producer producer = pulsarClient.createProducer(topicName, producerConf);
+        AtomicBoolean runTest = new AtomicBoolean(true);
+        Thread t1 = new Thread(() -> {
+            RateLimiter r = RateLimiter.create(produceRate);
+            while (runTest.get()) {
+                r.acquire();
+                producer.sendAsync("Hello World".getBytes());
+                consumer.receiveAsync().thenAccept(message -> consumer.acknowledgeAsync(message));
+            }
+        });
+        t1.start();
+        Thread.sleep(60000); // one minute wait for stats to be updated
```

Comment:
Reduced the sleep to two second and manually call the updateRates function.
Reducing the sleep to one second causes the rate to be shown as 0.